### PR TITLE
Add `OnlineLookupClass` to abstract the network calls for online lookup

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://app.clubhouse.io/simpledotorg/story/3948/create-use-case-to-query-the-online-lookup-api-and-fetch-the-medical-record

### CODE REVIEW

**Comment by vinaysshenoy** [comment]
This is a pretty large changeset, but a large portion of it is basically factory code for tests, and mapping code for converting the payload into domain objects, so hopefully shouldn't be too much of a pain to review.

**Review by Honey14 — CHANGES_REQUESTED** [review]
LGTM. Left one comment. PTAL

File: app/src/sharedTest/java/org/simple/clinic/TestData.kt [hunk 1]
```diff
@@ -658,7 +658,7 @@ object TestData {
       facilityUuid: UUID = UUID.fromString("faec54dc-1c5d-4768-83c5-80e7f272f8fe"),
       creationFacilityUuid: UUID? = UUID.fromString("faec54dc-1c5d-4768-83c5-80e7f272f8fe"),
       status: Appointment.Status = Appointment.Status.random(),
-      cancelReason: AppointmentCancelReason = AppointmentCancelReason.random(),
+      cancelReason: AppointmentCancelReason? = AppointmentCancelReason.random(),
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/sharedTest/java/org/simple/clinic/TestData.kt:661]: The model itself allows it as a nullable field. The `TestData` factory method incorrectly had it defined as a non-nullable parameter.

**Review by Honey14 — APPROVED** [review]
LGTM

**Pull request merged**